src/new/wallet.js

```diff
@@ -2469,6 +2470,73 @@ class HathorWallet extends EventEmitter {
 
     return tx;
   }
+
+  /**
+   * Queries the fullnode for a transaction
+   *
+   * @param {string} txId The transaction to query
+   *
+   * @returns {FullNodeTxResponse} Transaction data in the fullnode
+   */
+  async getFullTxById(txId) {
+    const tx = await new Promise(resolve => {
```

Comment:
About these interactions with `txApi`: 
- If it resolves with a success, it will call a callback and all goes well
- If it rejects, an `Unhandled promise exception` is thrown, which is expected to cause a crash in future versions of NodeJS when they make the default behavior for it stricter.

A code example of how to treat an exception here is available on the [`PartialTx` implementation](https://github.com/HathorNetwork/hathor-wallet-lib/blob/cf0ec6ea42169aed3c7f7fa6d3bb64649e83205e/src/models/partial_tx.ts#L498-L527), and I think a unit test would also be a good idea to ensure this crash is avoided on future refactors of the `TxApi` ( [existing code reference](https://github.com/HathorNetwork/hathor-wallet-lib/blob/cf0ec6ea42169aed3c7f7fa6d3bb64649e83205e/__tests__/models/partial_tx.test.js#L400-L414) as well ).

Reply:
Thanks, error handled in 395a7d3a3aa449c10d7fc6883405cba9473811b7